kernel/thread.c

```diff
@@ -167,61 +178,109 @@ void z_thread_monitor_exit(struct k_thread *thread)
 }
 #endif
 
-#ifdef CONFIG_THREAD_NAME
-void z_impl_k_thread_name_set(struct k_thread *thread, const char *value)
+int z_impl_k_thread_name_set(struct k_thread *thread, const char *value)
 {
+#ifdef CONFIG_THREAD_NAME
```

Comment:
Can't we just #if out the k_thread_name_set() API entirely if this isn't enabled instead of ging through the preprocessor hoops to make this return a runtime error?

Reply:
We could, but I think this is outside the scope of this patch, it was already like this, and this PR needs to be backported.